Source/WebCore/dom/Position.cpp

```diff
@@ -1143,8 +1153,10 @@ Position Position::leadingWhitespacePosition(Affinity affinity, bool considerNon
         return { };
 
     Position prev = previousCharacterPosition(affinity);
-    if (prev != *this && inSameEnclosingBlockFlowElement(deprecatedNode(), prev.deprecatedNode()) && is<Text>(*prev.deprecatedNode())) {
-        UChar c = downcast<Text>(*prev.deprecatedNode()).data()[prev.deprecatedEditingOffset()];
+    auto node = protectedDeprecatedNode();
```

Comment:
ditto.

Reply:
With my proposal it is clear if we're dealing with a Ref or a RefPtr. It is also more concise.